src/SupabaseClient.ts

```diff
@@ -94,20 +95,23 @@ export default class SupabaseClient<
     const defaultStorageKey = `sb-${new URL(this.authUrl).hostname.split('.')[0]}-auth-token`
     this.storageKey = options?.auth?.storageKey ?? defaultStorageKey
 
-    const settings = { ...DEFAULT_OPTIONS, ...options, storageKey: this.storageKey }
+    const settings = { ...DEFAULT_OPTIONS?.general, ...options }
 
     this.headers = { ...DEFAULT_HEADERS, ...options?.headers }
-    this.shouldThrowOnError = settings.shouldThrowOnError || false
 
-    this.auth = this._initSupabaseAuthClient(settings.auth || {}, this.headers, settings.fetch)
+    this.auth = this._initSupabaseAuthClient(
+      settings.auth || DEFAULT_OPTIONS.auth,
+      this.headers,
+      settings.fetch
+    )
     this.fetch = fetchWithAuth(supabaseKey, this._getAccessToken.bind(this), settings.fetch)
 
     this.realtime = this._initRealtimeClient({ headers: this.headers, ...settings.realtime })
     this.rest = new PostgrestClient(`${_supabaseUrl}/rest/v1`, {
       headers: this.headers,
       schema: options?.db?.schema,
       fetch: this.fetch,
-      throwOnError: this.shouldThrowOnError,
+      throwOnError: options?.db?.shouldThrowOnError || false,
```

Comment:
options?.db?.shouldThrowOnError ?? false

Here it is okay cos the second argument is false but better to use ??